### PR TITLE
Prices and volumes

### PR DESCRIPTION
**Pull request by GabrielCamba**
- Added conditional deployment for consuming prices from honeyswap on gc and uniswap v3 on mainnet and rinkeby
- Spreading prices to obtain volumes
- Calculating pairs as requested by coingecko

### CODE REVIEW

**Comment by anxolin** [comment]
@GabrielCamba Is this PR deployed somewhere too?

File: src/modules/tokens.ts
Comment (alfetopito) [inline src/modules/tokens.ts:42]: Curiosity: why can xdai have token prices while mainnet/rinkeby starts with 0?

File: src/utils/constants.ts
Comment (alfetopito) [inline src/utils/constants.ts:20]: Could you leave a comment explaining this?
Why you only add `xdai` to both of these Maps and they have different addresses

**Review by alfetopito — COMMENTED** [review]
Some general comments and questions.

As Anxo mentioned, would be great to have a link to the deployed graphs in the PR with test steps to validate the changes 👍

File: package.json
Comment (GabrielCamba) [reply package.json:14]: that's for supporting uniswap v3 implementation.

File: src/modules/tokens.ts
Comment (GabrielCamba) [reply src/modules/tokens.ts:42]: I'm not sure if I understood the question, but I added zero as a default on all tokens due to zero is neutral for addition.

File: src/uniswapMappings/uniswapPoolFactory.ts
Comment (GabrielCamba) [reply src/uniswapMappings/uniswapPoolFactory.ts:38]: that's inherited from uniswapV3 https://github.com/Uniswap/v3-subgraph/blob/bf03f940f17c3d32ee58bd37386f26713cff21e2/src/mappings/factory.ts#L56.
I can remove it or add something meaninfull. PLMK.

File: src/utils/constants.ts
Comment (GabrielCamba) [reply src/utils/constants.ts:20]: originally that mapping contained multiple netwoks, for xdai (gnosis chain), mainnet and rinkeby. So far we are using it only on gnosis chain that's why there's only xdai. I'll create a ticket for replacing it on next pr.

File: src/utils/pricing.ts
Comment (GabrielCamba) [reply src/utils/pricing.ts:61]: Yes, that's it. Another inherited comment. I'll remove it.

File: abis/UniswapV2Pair.json
Comment (GabrielCamba) [reply abis/UniswapV2Pair.json:713]: will create a task for this, for organizing ABIS

File: src/utils/constants.ts
Comment (anxolin) [reply src/utils/constants.ts:20]: I was also super confused by this, because it feels like things from other networks are misssing. 
Maybe you could at least explain why you don't need to set the values for other networks. Why is not used

File: src/utils/getPrices.ts
Comment (anxolin) [inline src/utils/getPrices.ts:74]: any non trivial code logic could use a simple comment of what u try to do

File: src/utils/pricing.ts
Comment (anxolin) [inline src/utils/pricing.ts:36]: pls, add comments too

Also, this list are not by network. How does this work for Gnosis chain and rinkebu for example

File: src/utils/pricing.ts
Comment (anxolin) [inline src/utils/pricing.ts:44]: do you know why 52?

File: src/utils/staticTokenDefinition.ts
Comment (anxolin) [inline src/utils/staticTokenDefinition.ts:25]: I understand you got this from Uniswap, and they implemented it for some reason. It would be nice to comment what was this reason, and point to their code as a reference

**Review by anxolin — APPROVED** [review]
Approving, because i want this merge ASAP, and I don't want you to add anything new in this PR, is too hard to review. Probably we have missed many things in our review because of this.  

Review the comments for following PRs

File: src/modules/tokens.ts
Comment (alfetopito) [reply src/modules/tokens.ts:42]: I mean, I understood that only gchain prices were initiated with `0` given the conditional

File: src/modules/pairs.ts
Comment (GabrielCamba) [reply src/modules/pairs.ts:14]: sadly yes. I wrote it in that way to prevent an error.

File: src/modules/pairs.ts
Comment (GabrielCamba) [reply src/modules/pairs.ts:53]: I understand that but some times some values are lost when changing stuff in the same line, depending on the type. It happens mostly with arrays.  That's why I wrote this. I know it's harder to spot errors and that's one of the reasons why I wanted to run tests, to check in the same line will work as expected. I can change it but I'll need some time to make it index to be sure it's not failing.

File: src/modules/tokens.ts
Comment (GabrielCamba) [reply src/modules/tokens.ts:33]: Network is gnosis chain, but it's name is taken from the subgraph deployment variable they use for gnosis-chain. It's still being called xdai, that's why the name.

File: src/utils/constants.ts
Comment (GabrielCamba) [reply src/utils/constants.ts:17]: already created a ticket for solving this-

File: src/utils/pricing.ts
Comment (GabrielCamba) [reply src/utils/pricing.ts:44]: no idea, Will investigate

File: src/modules/tokens.ts
Comment (GabrielCamba) [reply src/modules/tokens.ts:34]: will do.

File: src/utils/getPrices.ts
Comment (GabrielCamba) [reply src/utils/getPrices.ts:22]: not needed. getPair function will always return the address of the pool. 
https://docs.uniswap.org/protocol/V2/reference/smart-contracts/factory#getpair

File: src/utils/pricing.ts
Comment (GabrielCamba) [reply src/utils/pricing.ts:36]: copied code from here: https://github.com/Uniswap/v3-subgraph/blob/main/src/utils/pricing.ts

File: src/utils/pricing.ts
Comment (GabrielCamba) [reply src/utils/pricing.ts:44]: found it was changed 2 months ago: https://github.com/Uniswap/v3-subgraph/blob/bf03f940f17c3d32ee58bd37386f26713cff21e2/src/utils/pricing.ts#L45

**Comment by GabrielCamba** [comment]
> @GabrielCamba Is this PR deployed somewhere too?

https://thegraph.com/hosted-service/subgraph/gabrielcamba/cow-prices-mainnet?version=current
https://thegraph.com/hosted-service/subgraph/gabrielcamba/cow-prices-gc?version=current

File: src/utils/pricing.ts
Comment (anxolin) [reply src/utils/pricing.ts:36]: My comments is if you can add it in the code

File: src/utils/constants.ts
Comment (GabrielCamba) [reply src/utils/constants.ts]: moved map to a single variable.

File: src/utils/getPrices.ts
Comment (GabrielCamba) [reply src/utils/getPrices.ts:77]: done

File: src/modules/settlements.ts
Comment (anxolin) [reply src/modules/settlements.ts:28]: Link issue

File: src/utils/pricing.ts
Comment (anxolin) [reply src/utils/pricing.ts:44]: Nice to add a comment, with the link to the code. Probably this was a requirement like "lets consider pulls of $100K or more" but is expressed in ETH.

**Review by anxolin — APPROVED** [review]

**Review by alfetopito — APPROVED** [review]
As mentioned before, let's merge this and address any pending issues as follow up PRs

**Pull request merged**
